example/src/main/java/ably/io/tutorial/ExampleActivity.java

```diff
@@ -26,10 +32,47 @@ protected void onCreate(Bundle savedInstanceState) {
         } catch (AblyException e) {
             e.printStackTrace();
         }
+
+        /* set a button click listener for publishing messages */
+        findViewById(R.id.btPublish).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                /* Always do network instructions outside the Main Thread */
+                new Thread(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            /* Wrap method in try/catch block to prevent application from crashing due to AblyException */
+                            publishMessages();
+                        } catch (AblyException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }).start();
+
+            }
+        });
     }
 
     /* Add AblyException to method signature as AblyRest constructor can throw one */
     private void initAbly() throws AblyException {
         AblyRest ablyRest = new AblyRest(API_KEY);
+        /* Get channel for storing sounds */
+        channel = ablyRest.channels.get("persisted:sounds");
     }
+
+    private void publishMessages() throws AblyException {
+        Log.v(getClass().getCanonicalName(), "Publishing three messages...");
+
+        /* Get a null check in case ExampleActivity#initAbly method threw an exception */
+        if (channel != null) {
```

Comment:
I don't think we should have this check, if `initAbly` throws an exception then everything is broken anyway.  This will just fail silently and lead to more confusion.